soscleaner/soscleaner.py

```diff
@@ -423,6 +426,7 @@ def _prime_userdb(self):
 
         try:
             new_user = "obfuscateduser%s" % randint(1,1000000)
+            self.user_db['root'] = new_user
```

Comment:
I removed this via commit 4a7f10bba62cc1ad0523e05c3da75c023e5fc90b because I have modified soscleaner to only obfuscate users in the 1000-60000 UID ranges (as per /etc/login.defs) which are more propitious to be users created by users. I think soscleaner should focus on that instead of sanitizing everything

I took that route, after double-checking with some senior security folks and they were all agreed that spending time obfuscating what is obviously known and public (root, bin, www-data, ...) is not necessary. Additionally, it was generating a lot of false positive. So instead of maintaining a humongous ignore list, I prefered to have sosreport split the last output in ranges so that soscleaner can only rely on what it wants to obfuscate.

Since the introduction of 30d1b963e5b43ac03e308fec0c14b7046e2069b3, soscleaner now relies on `sos_commands/last/lastlog_-u_1000-60000`

```
-        self.users_file = 'sos_commands/last/lastlog'
+        self.users_file = 'sos_commands/last/lastlog_-u_1000-60000'
```
The corresponding sosreport change:
https://github.com/sosreport/sos/commit/a08196b8c02924a47271d6d31660b1ad528f09dd

With that being said, I removed 'root' because otherwise it will be obfuscated implicitly, and so far I didn't observe any regression due to this removal.

Reply:
Is there a way to make the test units happy without re-introducing this behaviour ?
Just thinking out loud here, can we pick a random username that we are sure will never get caught ?

Reply:
Sure.

That's a pretty big change though. What did the current large users say? @skhobbs18 represents probably the largest single userbase. Did he weigh in?

Reply:
I agree its probably not necessary to obfuscate known usernames, however, I feel my customer base would probably still want everything done.  How much processing time are we actually saving with this change?

Reply:
It's not much the processing but the false positive that obfuscating user in the 0-1000 UID range generate (at least for debian/ubuntu) since it may contains user such as : 'bin', 'sys', ....

where it could be found in a path, a file, a debian package, .... (basically all over the place).

Reply:
And at the end of the day, questioning some customers in the field, they are more interested in keep secret their own sauce more than what is obviously public.

Even if you hide root in the 'ps' output for something like this:
root         1     0  0 Nov13 ?        00:00:37 /sbin/init splash

It is safe to say that one can easily link the dots and figure out that the hidden user is root.
And if one really want to hide some public users, they can use the keyword obfuscation and that should do the trick.

Reply:
Understandable.  I will explain this to my customer should they complain about it.